internal/compiler/generator/rust.rs

```diff
@@ -796,6 +838,33 @@ fn generate_sub_component(
                     slint::re_exports::Orientation::Vertical => #layout_info_v,
                 }
             }
+
+            fn subtree_range(self: ::core::pin::Pin<&Self>, dyn_index: usize) -> slint::re_exports::IndexRange {
+                #![allow(unused)]
+                use slint::re_exports::*;
+                let _self = self;
+                match dyn_index {
+                    #(#repeated_subtree_ranges)*
+                    _ => panic!("invalid dyn_index {}", dyn_index),
+                }
+            }
+
+            fn subtree_component(self: ::core::pin::Pin<&Self>, dyn_index: usize, subtree_index: usize, result: &mut slint::re_exports::ComponentWeak) {
+                #![allow(unused)]
+                use slint::re_exports::*;
+                let _self = self;
+                match dyn_index {
+                    #(#repeated_subtree_components)*
+                    _ => panic!("invalid dyn_index {}", dyn_index),
+                };
+            }
+
+            fn index_property(self: ::core::pin::Pin<&Self>) -> usize {
```

Comment:
It looks to me that this function has only one call site. Maybe it should be inlined there?

Reply:
I would like to postpone this: I do not trust my rust-generator magic enough to not break something unrelated when I do that:-)